lib/static/modules/reducers/tree/helpers.js

```diff
@@ -69,3 +73,74 @@ export function resolveUpdatedStatuses(results, imagesById, suites) {
         }
     }
 }
+
+/**
+ *
+ * @note only one of "(imageIdsArray, newStatus)" or "images" need to be specified
+ * @param {Object} param0
+ * @param {{id: String, status: String}[]} param0.images
+ * @param {String[]} param0.imageIdsArray
+ * @param {String} param0.newStatus
+ */
+export function updateImagesStatus({tree, view, images, imageIdsArray, newStatus, diff = tree}) {
```

Comment:
Full cycle of updating image status.

- updates result statuses
- updates suite statuses
- calculates browser/images/suites showness/opennes

Reply:
100 lines of code just to update the statuses, pain...

To clarify, the problem is not in your code, but in the complexity of storage tests tree.

Reply:
Yeah, that's why i pulled the function out to "helpers"
But its still a lot of code, even when it is written almost imperatively: `updateParentSuitesStatus`, `calcSuitesOpenness`, `calcBrowsersOpenness`, `calcImagesOpenness`, `calcBrowsersShowness`, `calcSuitesShowness`